runtime/lua/vim/diagnostic.lua

```diff
@@ -399,17 +397,56 @@ local function set_list(loclist, opts)
 end
 
 ---@private
---- To (slightly) improve performance, modifies diagnostics in place.
-local function clamp_line_numbers(bufnr, diagnostics)
-  local buf_line_count = vim.api.nvim_buf_line_count(bufnr)
-  if buf_line_count == 0 then
-    return
+local function get_diagnostics(bufnr, opts, clamp)
```

Comment:
Assuming for qflist we only need to clamp line number and not col number?

Reply:
We're not doing any clamping for qflist right now (though maybe we should). The only things that go through this code path are the navigation methods, `open_float()`, and `show()`.

Reply:
I thought the call chain was:

set_qflist -> set_list -> M.get -> get_diagnostics with clamp == false

Reply:
Ah sorry, I misunderstood your comment. I think we probably should clamp for qflist as well.